proto/src/lib.rs

```diff
@@ -287,36 +287,50 @@ pub mod runtime {
         Ok((runtime, runtime_client))
     }
 
+    /// If the calling crate is cargo-shuttle, try to install shuttle-runtime from
+    /// a local version if this library is compiled in debug mode or from the production
+    /// branch of the shuttle repo if it's compiled in release mode.
+    ///
+    /// If the calling crate is deployer, use the version of shuttle-runtime installed in
+    /// deployer/prepare.sh.
+    ///
+    /// **Important:** When called by deployer, this function expects that shuttle-runtime
+    /// was installed in prepare.sh.
     fn get_runtime_executable() -> PathBuf {
-        // When this library is compiled in debug mode with `cargo run --bin cargo-shuttle`,
-        // install the checked-out local version of `shuttle-runtime
-        if cfg!(debug_assertions) {
-            // Path to cargo-shuttle
-            let manifest_dir = env!("CARGO_MANIFEST_DIR");
-
-            // Canonicalized path to shuttle-runtime
-            let path = std::fs::canonicalize(format!("{manifest_dir}/../runtime"))
-                .expect("failed to canonicalize path to runtime");
-
-            Command::new("cargo")
-                .arg("install")
-                .arg("shuttle-runtime")
-                .arg("--path")
-                .arg(path)
-                .output()
-                .expect("failed to install the shuttle runtime");
-        // When this library is compiled in release mode with `cargo install cargo-shuttle`,
-        // install the latest released `shuttle-runtime`
-        } else {
-            Command::new("cargo")
-                .arg("install")
-                .arg("shuttle-runtime")
-                .arg("--git")
-                .arg("https://github.com/shuttle-hq/shuttle")
-                .arg("--branch")
-                .arg("production")
-                .output()
-                .expect("failed to install the shuttle runtime");
+        let current_crate = env!("CARGO_PKG_NAME");
+
+        if current_crate == "cargo-shuttle" {
```

Comment:
Shouldn't it rather check if the runtime is already installed and just skip if it is?

Reply:
We'll just have to make sure to install the shuttle-runtime then when making changes to it in local dev, but that may be better than it always using the latest by force anyway. I'll implement this.

Reply:
Hmm, maybe we should then only check when in the `release` profile :man_shrugging:

Reply:
Yes, this will also prevent the deployer from updating the runtime unexpectedly. To update it now with the check, the user will need to `rm` and `new` their project, so it will use the new image.

Reply:
Perhaps what we should do is check that it is the same version as cargo-shuttle, and if it isn't then try to install? Because currently it will always try to install when in release mode, which may cause runtime to be updated before cargo-shuttle. If we change it to not install if it already is installed, the user will have to delete shuttle-runtime to update it.